lib/jsonapi_compliable/resource/configuration.rb

```diff
@@ -11,57 +42,222 @@ class << self
         class_attribute :adapter,
           :model,
           :type,
+          :serializer,
           :default_page_size,
-          :default_sort
+          :default_sort,
+          :attributes_readable_by_default,
+          :attributes_writable_by_default,
+          :attributes_sortable_by_default,
+          :attributes_filterable_by_default,
+          :relationships_readable_by_default,
+          :relationships_writable_by_default
+
+        # automodel
+        # autofilter
+        # autosort
+        # refactor finder tests to add serializer etc
+        #
+        # wrap restrict_filters, restrcit_sort
+        # restrict_filters_if
+        # restrict_sort_if
 
-        self.adapter ||= Adapters::Abstract.new
-        self.default_sort ||= []
-        self.default_page_size ||= 20
-        self.type ||= :undefined_jsonapi_type
+        class << self
+          prepend Overrides
+        end
+
+        def self.inherited(klass)
+          super
+          klass.config = Util::Hash.deep_dup(config)
+          klass.adapter ||= Adapters::Abstract.new
+          klass.default_sort ||= []
+          klass.default_page_size ||= 20
+          # re-assigning causes a new Class.new
+          if klass.serializer
+            klass.serializer = klass.serializer
+          else
+            klass.serializer = JSONAPI::Serializable::Resource
+          end
+          klass.type ||= klass.infer_type
+          default(klass, :attributes_readable_by_default, true)
```

Comment:
TODO: When running the generator, we should create an ApplicationResource with `*writeable` and `filterable` _at least_ defaulted to false.

Reply:
My thought is, we'd expose them all explicitly in ApplicationResource but default them all to `true`. I understand and agree with your thought here, but I think we have too much evidence showing this violates principle of least surprise for folks.